src/components/WorkOrders/UpdateJobForm.js

```diff
@@ -0,0 +1,110 @@
+import PropTypes from 'prop-types'
+import { GridRow, GridColumn } from '../Layout/Grid'
+import { Button, TextInput } from '../Form'
+import { useForm } from 'react-hook-form'
+import { useState } from 'react'
+import ExtraSorCode from './ExtraSorCode'
+
+const UpdateJobForm = ({
+  task,
+  sorCodes,
+  onGetToSummary,
+  sorCodesCollection,
+}) => {
+  const { register, handleSubmit, errors } = useForm()
+  const isContractorUpdatePage = true
+  const [temporarySorCodeCollection, setTemporarySorCodeCollection] = useState([
+    ...sorCodesCollection,
+  ])
+  const [nextFreeIndex, setNextFreeIndex] = useState(sorCodesCollection.length)
+
+  const onDeleteTemporarySorCode = (index) => {
+    let filtered = temporarySorCodeCollection.filter((e) => e.id != index)
+    setTemporarySorCodeCollection([...filtered])
+  }
+
+  const onAddTemporarySorCode = () => {
+    temporarySorCodeCollection.push({ id: nextFreeIndex })
+    setNextFreeIndex(nextFreeIndex + 1)
+    setTemporarySorCodeCollection([...temporarySorCodeCollection])
+  }
+
+  return (
+    <>
+      <form
+        role="form"
+        id="repair-request-form"
+        onSubmit={handleSubmit(onGetToSummary)}
+      >
+        <GridRow className="sor-code-select align-items-center">
+          {task.map((t, index) => (
+            <>
+              <GridColumn key={`gridcol1-${index}`} width="two-thirds">
+                <TextInput
+                  key={`gridcol1-textinput-${index}`}
+                  name={`sor-code-${index}`}
+                  label={index == 0 ? 'SOR code' : ''}
+                  widthClass={`sor-code-${index} govuk-!-width-full`}
+                  register={register}
+                  readOnly={true}
+                  value={`${t.code} ${
+                    t.description && t.description !== 'string'
+                      ? '- ' + t.description
+                      : ''
+                  }`}
```

Comment:
Bit confused here. What is this doing?

Reply:
### regarding "string" the same as before some will come with description = "string" 
and  existing tasks are a bit inconsistent, some come as code and description in `code` field  and some with code in code field and description in `description` field